### PR TITLE
Fix offset and animation issues

### PR DESCRIPTION
**Pull request by BPerlakiH**
Fixes: #815 
Fixes: #816 

It also fixes some other scroll issues.

## it was not possible to reveal the bars in certain cases:

https://github.com/kiwix/kiwix-apple/assets/6784320/9a36f06c-5ce5-457c-b8d7-a72da21fbf4e

## it was not possible to get out of a screen (forever flickering):

https://github.com/kiwix/kiwix-apple/assets/6784320/392015f2-3bf4-4113-b3bc-a00e059ce35d

### CODE REVIEW

**Comment by kelson42** [comment]
@BPerlakiH Thank you for the patch. Could you please explain briefly why the bug? Why only on iPad? How the solution looks like, because all of this is not clear to me.

**Comment by BPerlakiH** [comment]
Briefly it is connected to the scroll down to hide bars functionality: https://github.com/kiwix/kiwix-apple/pull/771
Which we applied only for iOS, but it meant to be really fully applicable only for iPhones. Whereas the implementation is shared between iPhone and iPads, it had some side effects for iPads, that is now fixed.

The offset of the initial screen appeared, due to this extra call we had:
```
webView.setValue(true, forKey: "_haveSetObscuredInsets")
```
which is in collision with hiding / showing the bars.

The other issue was that hiding / showing the bars in itself can cause a scroll event, which in return might toggle the bars again, and we can end up in an endless loop (as on the video).
In short solution to that: the hide / show of the bars is animated. We won't apply showing / hiding the bars until the previous animation is still running.

Another small fix is that, when the bottom bar is displayed, that is taken into consideration for laying out the web-view content (which wasn't the case previously).

**Review by rgaudin — APPROVED** [review]
LGTM
